libs/accounts/rate-limit/src/lib/rate-limit.ts

```diff
@@ -343,4 +350,145 @@ export class RateLimit {
     ]);
     return bans;
   }
+
+  /** search for rate limit statuses in redis by ip, uid, and email */
+  async search(
+    ip?: string,
+    email?: string,
+    uid?: string
+  ): Promise<BlockStatus[]> {
+    const identifiers: BlockOnOpts = {
+      ip,
+      email,
+      uid,
+      ip_email: ip || email ? `${ip || '*'}_${email || '*'}` : undefined,
+      ip_uid: ip || uid ? `${ip || '*'}_${uid || '*'}` : undefined,
+    };
+
+    return (
+      await Promise.all(
```

Comment:
Just make these direct calls. Thre's not enough things here to merit loops, and it'll be way more readable.